src/frontend/src/handler/mod.rs

```diff
@@ -49,12 +57,40 @@ mod show;
 pub mod util;
 pub mod variable;
 
+/// The [`PgResponse`] used by Risingwave.
+pub type RwPgResponse = PgResponse<PgResponseStream>;
+
+#[pin_project(project = PgResponseStreamImpl)]
+pub enum PgResponseStream {
+    LocalQuery(LocalQueryStream),
+    DistributedQuery(DistributedQueryStream),
+    Rows(BoxStream<'static, RowSetResult>),
+}
+
+impl Stream for PgResponseStream {
+    type Item = std::result::Result<Vec<Row>, BoxedError>;
+
+    fn poll_next(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
+        match self.project() {
```

Comment:
This is interesting. So without the pin_project crate we cannot do the matching here? What is the reason behind the scene?🤔

Reply:
It turns out matching can be done without pin_project. I had a wrong understanding of `Pin` previously.